### PR TITLE
[ReplaceGetEntityManagerToGetObjectManagerRector] Add rule and tests

### PR DESCRIPTION
**Pull request by JohJohan**
See #139

### CODE REVIEW

File: tests/Set/DoctrineORM213Set/Fixture/replace.php.inc [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Rector\Doctrine\Tests\Set\DoctrineORM213Set\Fixture;
+
+use Doctrine\Common\EventSubscriber;
+use Doctrine\ORM\Event\OnFlushEventArgs;
```
Comment (alexander-schranz) [inline tests/Set/DoctrineORM213Set/Fixture/replace.php.inc:6]: The test runs on `use Doctrine\ORM\Event\OnFlushEventArgs` but the rule goes on `Doctrine\ORM\Event\PreFlushEventArgs`.

Looks there are some more Classes needed to the ruleset which has `getEntityManager` deprecated.

File: config/sets/doctrine-orm-213.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+use Rector\Renaming\Rector\MethodCall\RenameMethodRector;
+use Rector\Renaming\ValueObject\MethodCallRename;
+
+return static function (RectorConfig $rectorConfig): void {
+    $rectorConfig->ruleWithConfiguration(
+        RenameMethodRector::class,
+        [
+            // @see https://github.com/doctrine/orm/issues/9875
+            new MethodCallRename(
+                'Doctrine\ORM\Event\PreFlushEventArgs',
```
Comment (alexander-schranz) [inline config/sets/doctrine-orm-213.php:15]: - OnClearEventArgs
- LifecycleEventArgs
- OnFlushEventArgs
- PostFlushEventArgs
- PreFlushEventArgs

Seems also have this deprecated method and should also be added?

Comment (JohJohan) [reply config/sets/doctrine-orm-213.php]: I added these events as well as test fixtures

**Comment by JohJohan** [comment]
Note i also replace getEntity to getObject

File: config/sets/doctrine-orm-213.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    $rectorConfig->ruleWithConfiguration(
+        RenameMethodRector::class,
+        [
+            // @see https://github.com/doctrine/orm/issues/9875
+            new MethodCallRename('Doctrine\ORM\Event\LifecycleEventArgs', 'getEntityManager', 'getObjectManager',),
+            new MethodCallRename('Doctrine\ORM\Event\OnClearEventArgs', 'getEntityManager', 'getObjectManager',),
+            new MethodCallRename('Doctrine\ORM\Event\OnFlushEventArgs', 'getEntityManager', 'getObjectManager',),
+            new MethodCallRename('Doctrine\ORM\Event\PostFlushEventArgs', 'getEntityManager', 'getObjectManager',),
+            new MethodCallRename('Doctrine\ORM\Event\PreFlushEventArgs', 'getEntityManager', 'getObjectManager',),
+            new MethodCallRename('Doctrine\ORM\Event\LifecycleEventArgs', 'getEntity', 'getObject',),
```
Comment (alexander-schranz) [inline config/sets/doctrine-orm-213.php:20]: Is duplicated

File: src/PhpDocParser/DoctrineClassAnnotationMatcher.php [hunk 4]
```diff
@@ -9,8 +9,9 @@
 
 final class DoctrineClassAnnotationMatcher
 {
-    public function __construct(private readonly ClassAnnotationMatcher $classAnnotationMatcher)
-    {
+    public function __construct(
+        private readonly ClassAnnotationMatcher $classAnnotationMatcher
+    ) {
```
Comment (alexander-schranz) [inline src/PhpDocParser/DoctrineClassAnnotationMatcher.php:14]: This change looks unexpected

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
I'm not a maintainer here and sadly can not start the CI tests here. Added just a few comments.

File: config/sets/doctrine-orm-213.php (hunk 3 above)
Comment (JohJohan) [reply config/sets/doctrine-orm-213.php:20]: Its not a duplicate this is replacing getEntity and https://github.com/rectorphp/rector-doctrine/pull/140/files#diff-53928e3609fdbbda67d3b1edc99d2830ef64f850af729b8eed4f26eebb85e392R15 replaces getEntityManager

File: src/PhpDocParser/DoctrineClassAnnotationMatcher.php (hunk 4 above)
Comment (JohJohan) [reply src/PhpDocParser/DoctrineClassAnnotationMatcher.php:14]: Correct will revert it

**Review by alexander-schranz — APPROVED** [review]

**Comment by alexander-schranz** [comment]
Looks fine from my side maybe @TomasVotruba approve the CI and take another look.

**Comment by samsonasik** [comment]
Please run:

```
vendor/bin/rector && composer fix-cs
```

to make CI green

**Comment by JohJohan** [comment]
@samsonasik i fixed specifis files when i run `vendor/bin/rector && composer fix-cs` it fixes allot more files then is said failing in the ci

**Comment by TomasVotruba** [comment]
The CI needs fixing. Check the failing job

**Review by samsonasik — APPROVED** [review]

File: config/sets/doctrine-orm-213.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+use Rector\Config\RectorConfig;
+use Rector\Renaming\Rector\MethodCall\RenameMethodRector;
+use Rector\Renaming\Rector\Name\RenameClassRector;
+use Rector\Renaming\ValueObject\MethodCallRename;
+
+return static function (RectorConfig $rectorConfig): void {
+    $rectorConfig->ruleWithConfiguration(
+        RenameMethodRector::class,
+        [
+            // @see https://github.com/doctrine/orm/issues/9875
```
Comment (TomasVotruba) [inline config/sets/doctrine-orm-213.php:14]: This link goes to some parent class renames. Is it correct?

Comment (JohJohan) [reply config/sets/doctrine-orm-213.php:14]: The deprecated class and functions where also linking to this page so I thought it was appropriate

Comment (TomasVotruba) [reply config/sets/doctrine-orm-213.php:14]: That issues seems WIP and unclear, what exactly is handles to me :)

This Rector rule renames method, so it should link commit/PR that renames method too. Could you link it?

Comment (JohJohan) [reply config/sets/doctrine-orm-213.php]: I added correct links

Comment (TomasVotruba) [reply config/sets/doctrine-orm-213.php]: Perfect, thank you :+1:

**Comment by TomasVotruba** [comment]
Thank you @JohJohan :clap:

**Comment by alexander-schranz** [comment]
Good work 👍

**Pull request merged**